tools/generate_test_status.py

```diff
@@ -20,57 +20,109 @@
 Prints {{id}} to terminal.
 """
 
-from utils import error, warning, metadata
+# The name of a job is HARDCODED.
+# If we change the name of a job in the test-all YML file,
+# or use a different prefix from a caller (like the test-all-and-report YML file),
+# then we need to adjust the hardcoded names in this python script.
+
+from utils import error, warning, to_bool, normalize_pkg_name, metadata
 
 import sys
 import os
 import glob
 import json
 from datetime import datetime
+import requests
 
 from typing import Any, Dict
 
 ################################################################################
 # Arguments
 num_args = len(sys.argv)
 
-if num_args > 5:
-    error("Too many arguments")
-
-repo = runID = hash = which_gap = "Unknown"
+if num_args != 7:
+    error("Unknown number of arguments")
 
-if num_args > 1:
-    repo = "https://github.com/" + sys.argv[1]
-if num_args > 2:
-    runID = sys.argv[2]
-if num_args > 3:
-    hash = sys.argv[3]
-if num_args > 4:
-    which_gap = sys.argv[4]
+git_token = sys.argv[1]
+repo = sys.argv[2]
+run_id = sys.argv[3]
+hash = sys.argv[4]
+which_gap = sys.argv[5]
+is_workflow_call = to_bool(sys.argv[6])
 
 ################################################################################
-# Collect the job-status of each package from _reports/
+# Collect names of all packages
 files = []
-for file in glob.glob("_reports/**/*.json", recursive=True):
+for file in glob.glob("packages/*/meta.json"):
     files.append(file)
 
 files.sort()
-
-pkgs = {}
+pkgs: Dict[str, Any] = {}
 
 for file in files:
-    with open(file, "r", encoding="utf-8", errors="ignore") as f:
-        data = json.load(f)
+    pkgs[normalize_pkg_name(file)] = {}
 
-    pkgs[os.path.splitext(os.path.basename(file))[0]] = data
+################################################################################
+# Collect job information for all packages:
+# Retrieve additional meta data about this workflow run via the REST API.
+# We use this to figure out the status of all jobs
+# as well as the numeric ids of all jobs, which we need
+# to generate direct links to jobs in the final status report.
+# https://stackoverflow.com/questions/33878019/how-to-get-data-from-all-pages-in-github-api-with-python
+url = f"https://api.github.com/repos/{repo}/actions/runs/{run_id}/jobs?simple=yes&per_page=100&page=1"
+res = requests.get(url, headers={"Authorization": git_token})
+jobs = res.json()["jobs"]
+while "next" in res.links.keys():
+    res = requests.get(res.links["next"]["url"], headers={"Authorization": git_token})
+    jobs.extend(res.json()["jobs"])
+
+# direct link for skipped packages
+if is_workflow_call:
+    name = f"{which_gap} / Build GAP and packages"
+else:
+    name = "Build GAP and packages"
```

Comment:
Maybe instead of passing in an `is_workflow_call` boolean, pass in a `job_name_prefix` which is either empty or `{which_gap} / `. Then this code just becomes
```suggestion
name = f"{job_name_prefix}Build GAP and packages"
```